client/core/core.go

```diff
@@ -2762,22 +2835,43 @@ func handleMatchRoute(c *Core, dc *dexConnection, msg *msgjson.Message) error {
 	if err != nil {
 		return fmt.Errorf("match request parsing error: %v", err)
 	}
+
+	// Acknowledgements MUST be in the same orders as the msgjson.Matches.
 	matches, acks, err := dc.parseMatches(msgMatches, true)
 	if err != nil {
+		// Even one failed match fails them all since the server requires acks
+		// for them all, and in the same order. TODO: consider lifting this
+		// requirement, which requires changes to the server's handling.
 		return err
 	}
-
 	resp, err := msgjson.NewResponse(msg.ID, acks, nil)
 	if err != nil {
 		return err
 	}
+
+	// Send the match acknowledgments.
+	// TODO: Consider a "QueueSend" or similar, but do not bail on the matches.
 	err = dc.Send(resp)
 	if err != nil {
-		return err
+		log.Errorf("Send match response: %v", err)
+		// dc.addPendingSend(resp) // e.g.
 	}
```

Comment:
I didn't like that a dropped connection or send failure for any other reason was preventing the matches from being run even though the server does need the match ack, and I still don't love that we don't have an outgoing message queue to retry sending such important responses since the client has no way presently to recover from this (the server won't send a match request again just because it timed out waiting for a response).  There's lots of room for improvement in this area.

Reply:
Right. This is documented under **Potential causes of trade disruptions** in #361 which is a little similar to a previously existing issue where failure to send an init or redeem request to the server would cause the trade to go to hell. The latter is fixed with `resendPendingRequests` and similar stuff can be done for request acks send errors. Can create an issue to track this.

Reply:
Nice, thanks @itswisdomagain